File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItOrderExpirationTypeParcelable.java
package it.trade.android.sdk.model;

import android.os.Parcel;
import android.os.Parcelable;

import it.trade.android.sdk.enums.TradeItOrderExpirationType;

public class TradeItOrderExpirationTypeParcelable implements Parcelable {
    private TradeItOrderExpirationType expiration;
    private String displayLabel;

    TradeItOrderExpirationTypeParcelable(TradeItOrderExpirationType expiration, String displayLabel) {
        this.expiration = expiration;
        this.displayLabel = displayLabel;
    }

    public TradeItOrderExpirationType getExpirationType() {
        return expiration;
    }

    public void setExpiration(TradeItOrderExpirationType expiration) {
        this.expiration = expiration;
    }

    public String getDisplayLabel() {
        return displayLabel;
    }

    public void setDisplayLabel(String displayLabel) {
        this.displayLabel = displayLabel;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TradeItOrderExpirationTypeParcelable that = (TradeItOrderExpirationTypeParcelable) o;

        if (expiration != that.expiration) return false;
        return displayLabel != null ? displayLabel.equals(that.displayLabel) : that.displayLabel == null;
    }

    @Override
    public int hashCode() {
        int result = expiration != null ? expiration.hashCode() : 0;
        result = 31 * result + (displayLabel != null ? displayLabel.hashCode() : 0);
        return result;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.expiration == null ? -1 : this.expiration.ordinal());
        dest.writeString(this.displayLabel);
    }

    protected TradeItOrderExpirationTypeParcelable(Parcel in) {
        int tmpExpiration = in.readInt();
        this.expiration = tmpExpiration == -1 ? null : TradeItOrderExpirationType.values()[tmpExpiration];
        this.displayLabel = in.readString();
    }

    public static final Parcelable.Creator<TradeItOrderExpirationTypeParcelable> CREATOR = new Parcelable.Creator<TradeItOrderExpirationTypeParcelable>() {
        @Override
        public TradeItOrderExpirationTypeParcelable createFromParcel(Parcel source) {
            return new TradeItOrderExpirationTypeParcelable(source);
        }

        @Override
        public TradeItOrderExpirationTypeParcelable[] newArray(int size) {
            return new TradeItOrderExpirationTypeParcelable[size];
        }
    };
}

File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItOrderExpirationTypeParcelable.kt
package it.trade.android.sdk.model

import android.os.Parcel
import android.os.Parcelable

import it.trade.android.sdk.enums.TradeItOrderExpirationType

class TradeItOrderExpirationTypeParcelable : Parcelable {
    var expirationType: TradeItOrderExpirationType? = null
        private set
    var displayLabel: String? = null

    internal constructor(expiration: TradeItOrderExpirationType, displayLabel: String) {
        this.expirationType = expiration
        this.displayLabel = displayLabel
    }

    fun setExpiration(expiration: TradeItOrderExpirationType) {
        this.expirationType = expiration
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val that = o as TradeItOrderExpirationTypeParcelable?

        if (expirationType != that!!.expirationType) return false
        return if (displayLabel != null) displayLabel == that.displayLabel else that.displayLabel == null
    }

    override fun hashCode(): Int {
        var result = if (expirationType != null) expirationType!!.hashCode() else 0
        result = 31 * result + if (displayLabel != null) displayLabel!!.hashCode() else 0
        return result
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(if (this.expirationType == null) -1 else this.expirationType!!.ordinal)
        dest.writeString(this.displayLabel)
    }

    protected constructor(`in`: Parcel) {
        val tmpExpiration = `in`.readInt()
        this.expirationType = if (tmpExpiration == -1) null else TradeItOrderExpirationType.values()[tmpExpiration]
        this.displayLabel = `in`.readString()
    }

    companion object {

        val CREATOR: Parcelable.Creator<TradeItOrderExpirationTypeParcelable> = object : Parcelable.Creator<TradeItOrderExpirationTypeParcelable> {
            override fun createFromParcel(source: Parcel): TradeItOrderExpirationTypeParcelable {
                return TradeItOrderExpirationTypeParcelable(source)
            }

            override fun newArray(size: Int): Array<TradeItOrderExpirationTypeParcelable?> {
                return arrayOfNulls(size)
            }
        }
    }
}
